src/screens/tabs/Profile/screens/Accounts/AddICNS/index.tsx

```diff
@@ -0,0 +1,167 @@
+import { t } from 'i18next';
+import React, {
+  RefObject,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import {
+  ActivityIndicator,
+  Linking,
+  TouchableOpacity,
+  View,
+} from 'react-native';
+import { Modalize } from 'react-native-modalize';
+import { useDispatch, useSelector } from 'react-redux';
+
+import RainbowButton from '@/components/buttons/RainbowButton';
+import {
+  ActionButton,
+  ActionSheet,
+  Header,
+  Image,
+  Modal,
+  Text,
+} from '@/components/common';
+import { Option } from '@/components/common/ActionSheet';
+import Icon from '@/components/icons';
+import { Colors } from '@/constants/theme';
+import { getICNSData, setReverseResolvedName } from '@/redux/slices/keyring';
+import { ICNS_LEARN_MORE_URL, ICNS_LOGO, ICNS_URL } from '@/services/ICNS';
+
+import styles from './styles';
+
+interface Props {
+  modalRef: RefObject<Modalize>;
+}
+
+const none = 'None';
+
+function AddICNS({ modalRef }: Props) {
+  const dispatch = useDispatch();
+  const [actionSheetOptions, setActionSheetOptions] = useState<Option[]>([]);
+  const actionSheetRef = useRef<Modalize>(null);
+  const { reverseResolvedName, names } = useSelector(
+    state => state.keyring.currentWallet?.icnsData
+  ) || { names: [] };
+  const { icnsDataLoading } = useSelector(state => state.keyring);
+  const [selectedName, setSelectedName] = useState<string>(
+    reverseResolvedName || none
+  );
+  const noNames = useMemo(() => !names.length, [names]);
+
+  const [reverseResolveLoading, setReverseResolveLoading] =
+    useState<boolean>(false);
+  const dataLoading = !reverseResolveLoading && icnsDataLoading;
+
+  const setICNSName = useCallback(() => {
+    setReverseResolveLoading(true);
+    dispatch(
+      setReverseResolvedName({
+        name: selectedName,
+        onFinish: () => {
+          modalRef.current?.close();
+        },
+      })
+    );
+  }, [selectedName]);
+
+  useEffect(() => {
+    dispatch(getICNSData());
+  }, []);
+
+  const handleOnPress = () => {
+    const noneOption = {
+      id: 0,
+      label: none,
+      onPress: () => setSelectedName(none),
+      icon: null,
+    };
+    const options = [
+      ...names.map((name: string, index: number) => ({
+        id: index + 1,
+        label: name,
+        onPress: () => setSelectedName(name),
+        icon: null,
+      })),
+      noneOption,
+    ];
+
+    setActionSheetOptions(options);
+    actionSheetRef?.current?.open();
+  };
```

Comment:
Do we need to generate the options in handleOnPress? We could do something like this

```suggestion
const options = useMemo(() => [
      ...names.map((name: string) => ({
        id: name,
        label: name,
        onPress: () => setSelectedName(name),
        icon: null,
      })),
      {
        id: none,
        label: none,
        onPress: () => setSelectedName(none),
        icon: null,
      },
  ], [names])

const handleOnPress = () => actionSheetRef?.current?.open();
```

Reply:
in the profile is done like that because the options depend on a param received by the function, but here we could do it like the suggestion